pyanaconda/core/util.py

```diff
@@ -1236,15 +1243,15 @@ def collect(module_pattern, path, pred):
                             sys.modules[module_part_name] = module_part
 
                     # load the collected module
-                    module = imp.load_module(module_pattern % mod_name,
-                                             fo, module_path, module_flags)
+                    module = importlib.util.module_from_spec(spec)
+                    sys.modules[mod_name] = module
+                    spec.loader.exec_module(module)
 
             # get the filenames without the extensions so we can compare those
             # with the .py[co]? equivalence in mind
             # - we do not have to care about files without extension as the
             #   condition at the beginning of the for loop filters out those
-            # - module_flags[0] contains the extension of the module imp found
-            candidate_name = module_path[:module_path.rindex(module_flags[0])]
+            # - found_ext contains the extension of the module importlib found
             loaded_name, loaded_ext = module.__file__.rsplit(".", 1)
```

Comment:
I think that this will remove the dot from the extension.

Reply:
Yes! 3 lines below: `loaded_ext = "." + loaded_ext`